Realm/RLMConfiguration.h

```diff
@@ -0,0 +1,65 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2015 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#import <Foundation/Foundation.h>
+#import <Realm/RLMRealm.h>
+
+RLM_ASSUME_NONNULL_BEGIN
+
+/**
+ An `RLMConfiguration` is used to describe the different options used to
+ create an `RLMRealm` instance.
+ */
+@interface RLMConfiguration : NSObject<NSCopying>
+
+/**
+ Returns the default configuration used to create Realms when no other
+ configuration is explicitly specified.
+
+ @return The default Realm configuration.
+ */
++ (instancetype)defaultConfiguration;
+
+/**
+ Sets the default configuration to the given `RLMConfiguration`.
+
+ @param configuration The new default Realm configuration.
+ */
++ (void)setDefaultConfiguration:(nullable RLMConfiguration *)configuration;
```

Comment:
Why is that preferable to having users pass `[[RLMConfiguration alloc] init]` as the argument, or providing a separate `-resetDefaultConfiguration` method.